jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/EcsExecUtils.kt

```diff
@@ -87,4 +96,22 @@ object EcsExecUtils : CoroutineScope by ApplicationThreadPoolScope("EcsExec") {
             }
         )
     }
+
+    fun ensureServiceIsInStableState(project: Project, service: Service, actionName: String, block: () -> Unit) {
+        val response = project.awsClient<EcsClient>().describeServices(
+            DescribeServicesRequest.builder()
+                .cluster(service.clusterArn())
+                .services(service.serviceArn()).build()
+        )
+        val serviceStateChangeInProgress = response.services().first().deployments().first().rolloutState() == DeploymentRolloutState.IN_PROGRESS
+        if (serviceStateChangeInProgress) {
+            if (actionName == message("ecs.execute_command_enable")) {
+                notifyWarn(actionName, message("ecs.execute_command_enable_in_progress", service.serviceName()), project)
+            } else {
+                notifyWarn(actionName, message("ecs.execute_command_disable_in_progress", service.serviceName()), project)
+            }
+        } else {
+            block()
+        }
```

Comment:
Using a localized string here as the == seems like a code smell to me

I would make this method return true or false and let the call site determine what to do

Reply:
Made the change